pkg/virt-config/feature-gates.go

```diff
@@ -74,6 +74,8 @@ const (
 	BochsDisplayForEFIGuests = "BochsDisplayForEFIGuests"
 	// NetworkBindingPlugingsGate enables using a plugin to bind the pod and the VM network
 	NetworkBindingPlugingsGate = "NetworkBindingPlugins"
+	// AutoMemoryLimitsGate enables automatic setting of vmi memory limits if there is a ResourceQuota with memory limits associated with the vmi namespace.
+	AutoMemoryLimitsGate = "AutoMemoryLimits"
```

Comment:
If we don't go with the annotation approach, I think that we at least need some more aggressive documentation here.

WDYT about adding something like:
```
// track: Alpha
// Alpha features provide no backward compatibility guarantees, may dramatically change in the
// future and are not thoroughly tested. They are intended to be used by advanced users at their own risk.
```

Reply:
TBH I am not sure! The [definition](https://kubevirt.io/user-guide/operations/activating_feature_gates/#activating-feature-gates) of a FG is exactly that it is not mature enough.
We could start thinking of something like this:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/features/kube_features.go#L27-L42
But I think it requires a discussion with the definition of the several steps.
WDYT?

Reply:
>We could start thinking of something like this:
>https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/features/kube_features.go#L27-L42
> But I think it requires a discussion with the definition of the several steps.

Yeah I guess you're right.
I'm currently working on a proposal on that front.